src/adagrams.js

```diff
@@ -1,15 +1,148 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+  }
+  
+  const letterPoolArray = []
+  const lettersInHand = []
+
+  for (let letter in letterPool) {
+    for (let i = 0; i < letterPool[letter]; i++) {
+      letterPoolArray.push(letter);
+    }
+  }
+
+  for (let i = 0; i < 10; i++) {
+    let randomNumber = [Math.floor(Math.random() * letterPoolArray.length)];
+    lettersInHand.push(letterPoolArray[randomNumber]);
+    letterPoolArray.splice(randomNumber, 1);
+  }
+
+  return lettersInHand
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  for (let letter of input.toUpperCase()) {
+    if (lettersInHand.includes(letter)) {
+      let letterInstance = lettersInHand.indexOf(letter);
+      lettersInHand.splice(letterInstance, 1);
+    } else {
+        return false;
+    }
+  }
+
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  const letterValue = {
+    'A': 1,
+    'E': 1,
+    'I': 1,
+    'O': 1,
+    'U': 1,
+    'L': 1,
+    'N': 1,
+    'R': 1,
+    'S': 1,
+    'T': 1,
+    'D': 2,
+    'G': 2,
+    'B': 3,
+    'C': 3,
+    'M': 3,
+    'P': 3,
+    'F': 4,
+    'H': 4,
+    'V': 4,
+    'W': 4,
+    'Y': 4,
+    'K': 5,
+    'J': 8,
+    'X': 8,
+    'Q': 10,
+    'Z': 10
+  }
+  let score = 0;
+
+  if (word.length == 0) {
+    return score;
+  } else {
+    for (let letter of word.toUpperCase()) {
+      score += letterValue[letter];
+    }
+    if (word.length >= 7) {
+      score += 8;
+    }
+  }
+
+  return score;
 };
 
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
-};
+  const wordScoreDict = {};
+  let highestScore = 0;
+  
+  for (let i = 0; i < words.length; i++) {
+    if (scoreWord(words[i]) in wordScoreDict) {
+      wordScoreDict[scoreWord(words[i])].push(words[i]);
+    } else {
+      wordScoreDict[scoreWord(words[i])] = [words[i]]
+    }
+
+    if (scoreWord(words[i]) > highestScore) {
+      highestScore = scoreWord(words[i])
+    }
+  }
+
+  let shortestWord = 'XXXXXXXXXX';
+
+  if (wordScoreDict[highestScore].length > 1) {
+    for (let word of wordScoreDict[highestScore]) {
+      if (word.length === 10) {
+        return {
+          'word': word,
+          'score': highestScore
+        }
+      } else {
+        if (word.length < shortestWord.length) {
+          shortestWord = word;
+        }
+      }
+    }
+    return {
+      'word': shortestWord,
+      'score': highestScore
+    }
+  }
+
+  return {
+    'word': wordScoreDict[highestScore][0],
+    'score': highestScore
+  }
```

Comment:
Nice job, Winslow! As you probably noticed, I didn't give too much feedback due to the scope of the project (Translating your python code to Javascript code). I tried to just point out any glaring issues, if any. With that being said, if you want to discuss anything in greater detail, feel free to reach out to me!